src/pages/dapp/index.tsx

```diff
@@ -227,20 +239,28 @@ export function StarRating(props) {
       onChange(rating);
     }
   }, [rating]);
+
+  const handleChangeRating = (ratings: number) => {
+    setHover(0);
+    setRating(ratings);
+  };
+
   return (
     <Row
       className="gap-x-[4px]"
-      onMouseLeave={editable ? () => setRating(rating) : undefined}
+      onMouseLeave={editable ? () => handleChangeRating(rating) : undefined}
```

Comment:
Instead of adding conditions in the function declaration could've moved to the function definition itself. It'll look cleaner and readable.